### ISSUE TITLE
Query log cosmetic changes

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdGuardHome#481]
Source: https://github.com/AdguardTeam/AdGuardHome/issues/480

> On the "Query Log" page, it would be great to be able to specify a time range, e.g. show only queries from 11:48:00 to 11:49:00
> Also, when I want to see more rows of data and select "100 rows" on the bottom of the screen, it would be great if AdGuard would remember that choice.

**Comment by tomiams** [comment]
While we're at it, it would be superb to
1. toggle live view while we're viewing the log.
2. Add more options than 100 rows

**Comment by ArchiveBase** [comment]
@ameshkov
I point users feature requests as summary and some details to implement this facility better way. Many people concern about manage logged data and view them perfectly also may helps to avoid unwanted codes and continuously make changes within various requests.

-----

Add a specific area (for example in query log page) to set,

   1. **Date range**
      Day/ Week/ Month/ Custom 
   _Custom - let user to pick any calendar date range. it can be between 3 days (24/July/2019 - 26/July/2019) Or 2 months (15/Jan/2019 - 18/Feb/2019)_
(related https://github.com/AdguardTeam/AdGuardHome/issues/750 , https://github.com/AdguardTeam/AdGuardHome/issues/452 , https://github.com/AdguardTeam/AdGuardHome/issues/776 )

  Together

   2. **Time range**
      5min/ 10min/ 1hr/ Custom
 _Custom - let user to pick any time range from 0.00 to 24.00 (AM/PM) It means as you mentioned above "e.g. show only queries from 11:48:00 to 11:49:00" and your another issue https://github.com/AdguardTeam/AdGuardHome/issues/314 might not necessary^^_

![Screenshot_4](https://user-images.githubusercontent.com/46082456/61992225-a3fe4600-b085-11e9-94f8-6fea4c2e73f3.png)
(classic sample)

_(Resources)
 https://www.ntop.org/products/traffic-analysis/ntop
 https://github.com/ntop/ntopng 
 https://www.glasswire.com_ 


   3. **Clients range** 
   (currently able view per client or all only)

       _If no range then must sort according to (1) and (2) and user able to enter clients range as 192.168.0.100 - 192.168.0.115 or as one by one 192.168.0.100,192.168.0.101, 192.168.0.105_

-----
   
Now result. 

Above **( 1 , 2 and 3 )** inputs on Query Page must apply to almost AGH UI components and keep remember[1] during session https://github.com/AdguardTeam/AdGuardHome/issues/672 unless user modify it (similar[1] as we filter issues on GitHub)

  _( To clear it should apply to every pages, Dashboard( top clients, queried/blocked domains, General counters https://github.com/AdguardTeam/AdGuardHome/issues/314 << ) Charts/Graph, any other various Statics such as used filters https://github.com/AdguardTeam/AdGuardHome/issues/895 and might useful for upcoming features etc._
  
While already sorted as we most expected then existing results still able sort further https://github.com/AdguardTeam/AdGuardHome/issues/882
 
_( cover https://github.com/AdguardTeam/AdGuardHome/issues/767 )_

-----
-----

"See more rows of data and select "100 rows" on the bottom of the screen, it would be great if AdGuard would remember that choice."
(above @ameshkov)

https://github.com/AdguardTeam/AdGuardHome/issues/855 , https://github.com/AdguardTeam/AdGuardHome/issues/802 , https://github.com/AdguardTeam/AdGuardHome/issues/845 ,

**Comment by ArchiveBase** [comment]
https://github.com/AdguardTeam/AdGuardHome/issues/944

**Comment by ArchiveBase** [comment]
Enhancement for sort under **'3. Clients range'**

Recently user had opened issue about "Client Groups" https://github.com/AdguardTeam/AdGuardHome/issues/962 then i see above explained details limited because re-check all devices belong to certain group and manually input them to sort quite annoying task also some devices may not assigned to static local IP address (usually added by MAC address)

So i prefer request to add easy option for sort by 'group name' (result must apply to Query log page and  Dashboard as well)

**Comment by ArchiveBase** [comment]
It's shame to AG that Pi-Hole has this feature😉 (today noticed that)

![Screenshot_1](https://user-images.githubusercontent.com/46082456/67645736-effaa880-f95c-11e9-8aea-dd538c776f2b.png)

**Comment by szolin** [comment]
More tasks for query logs:
* The Previous and Next buttons on top (pt.4 from #1125)
* Add a "Block client" button to the Query Log #1097 
* Query Log screen enhancement - Show name of Filters list #993
* Responsive Design on Query Log page (smartphone) #978 
* change querylog.json location or add live view only (no write to disk) #876
* Remember Row Count selection on Refresh #855
* Query Log - Basic/Detailed View #697
* Less Obtrusive Block/Unblock Button in Query Log #684
* Domain filter in query log is case-sensitive #680 
* Query Log - Compact Mode #622 
* Add filter option to the filtering log #594

**Comment by Archive5** [comment]
Other tasks for query logs:

- Add response time to Query Log #1601
- Allow searching in response field #1416

**Comment by lordraiden** [comment]
It would be nice as well to filter the dashboard view based on subnets (vlans) and by group of tags that are defined in adguard home
So I can have quick view of how is doing my LAN, IoT, server and DMZ networks in a separated way, for example.

**Comment by ammnt** [comment]
How about adding control characters for a more comfortable search?
https://en.wikipedia.org/wiki/Control_character

**Comment by ameshkov** [comment]
Do you mean extending search syntax so that you could specify additional search criteria right in the query text? Smth like what Google provides?

**Comment by ammnt** [comment]
> Do you mean extending search syntax so that you could specify additional search criteria right in the query text? Smth like what Google provides?

Yes, but rather as a search tool in text editors like Notepad and Sublime Text. It will be clearer to most people.

**Comment by Archive5** [comment]
> > Do you mean extending search syntax so that you could specify additional search criteria right in the query text? Smth like what Google provides?
> 
> Yes, but rather as a search tool in text editors like Notepad++ and Sublime Text. It will be clearer to most people.

#594 https://github.com/AdguardTeam/AdGuardHome/issues/594#issuecomment-469129412

**Comment by seonwoolee** [comment]
Could we have sorting by request type? Plain DNS, DNS over TLS, DNS over HTTPS, DNS over QUIC?

**Comment by ameshkov** [comment]
sort or filter by it?

**Comment by seonwoolee** [comment]
Oh whoops I meant filter

**Comment by ammnt** [comment]
@ainar-g & @ameshkov, please display the date or date+time combination of query instead of just time in compact mode. Otherwise, it is not clear when this request was actually made:

![image](https://user-images.githubusercontent.com/45385632/121074132-1b51a480-c7dc-11eb-88fe-364288a88663.png)

Thank you😣

**Comment by ainar-g** [comment]
Depends on #2290.

**Comment by thomaslacour** [comment]
Is this feature coming soon :) ?

**Comment by kamanwu** [comment]
really need this feature

**Comment by msbjr88** [comment]
So is this being worked or not?

**Comment by mbhs-scott** [comment]
Is there any timeline for this request at all?

**Comment by thibaultmeyer** [comment]
Any news about this must have Feature ?

**Comment by tjfriese** [comment]
It would be really nice if it were possible to filter the query logs by a date range.
